docs/source/audio_load.mdx

```diff
@@ -57,7 +57,41 @@ If you only want to load the underlying path to the audio dataset without decodi
 
 ## AudioFolder
 
-You can also load a dataset with an `AudioFolder` dataset builder. It does not require writing a custom dataloader, making it useful for quickly loading audio data.
+You can also load a dataset with an `AudioFolder` dataset builder. It does not require writing a custom dataloader, making it useful for quickly loading audio data. For classification tasks, your audio dataset structure should look like this:
+
+```
+folder/train/electronic/01.mp3
+folder/train/punk/01.mp3
+```
+
+Load your dataset by specifying `audiofolder` and the directory of your dataset in `data_dir`:
+
+```py
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("audiofolder", data_dir="/path/to/folder")
+>>> dataset["train"][0]
+{'audio': {'path': '/path/to/electronic/01.mp3',
+  'array': array([ 3.9714024e-07,  7.3031038e-07,  7.5640685e-07, ...,
+         -1.1963668e-01, -1.1681189e-01, -1.1244172e-01], dtype=float32),
+  'sampling_rate': 44100},
+ 'label': 0}
+
+>>> dataset["train"][-1]
+{'audio': {'path': '/path/to/punk/01.mp3',
+  'array': array([0.15237972, 0.13222949, 0.10627693, ..., 0.41940814, 0.37578005,
+         0.33717662], dtype=float32),
+  'sampling_rate': 44100},
+ 'label': 1}
+```
+
+If your directory contains multiple splits, use the `data_files` argument to load the images associated with each split:
```

Comment:
in fact it's not needed, the directory structure will be parsed if you provide data directory to `data_dir` too :) 
in case if different metadata files have different set of features, it would anyway throw an error (this is intentional), so the only way to load such cases is to load them separately (with different `load_dataset` i mean). if you think it's an incorrect behavior, feel free to open an issue and ping me and @mariosasko and @lhoestq, we'll discuss it :) 


note aside - you can also use `glob` style in data files: `data_files={"train":["train/electronic/**/*.mp3"]}`

Reply:
Thanks for the explanation! I think the behaviour makes sense, but I've added a small note here to help future users understand what to do: https://github.com/huggingface/datasets/pull/4925/files#diff-80eb88d203f3d89e6416bf139bf3caef42021d5953261f526fc656b9871325c9R148